packages/cinterop/src/jvm/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -138,7 +155,15 @@ actual object RealmInterop {
     }
 
     actual fun realm_open(config: NativePointer, dispatcher: CoroutineDispatcher?): NativePointer {
-        val realmPtr = LongPointerWrapper(realmc.realm_open((config as LongPointerWrapper).ptr))
+        val realmPtr: LongPointerWrapper = if (dispatcher != null) {
+            // create a custom Scheduler for JVM, use a lock to prevent concurrent modification of the RealmConfig
+            realmOpenLock.withLock {
```

Comment:
I guess we need to do this locking on Darwin too. Alternatively we could use an atomicfu lock higher up the stack.

Reply:
Darwin doesn't modify the Configuration so there's no need to lock